lib/rubygems/sigstore/rekor/hashed_rekord.rb

```diff
@@ -0,0 +1,6 @@
+require "rubygems/sigstore/rekor/log_entry"
+require "rubygems/sigstore/rekor/signature"
+
+class Gem::Sigstore::Rekor::HashedRekord < Gem::Sigstore::Rekor::LogEntry
```

Comment:
As far as our signature verification logic is concerned, hasked rekords and rekords look the same. I moved all of the signature code into a module. Neither `Rekord` nor `HashedRekord` have additional code at the moment.